src/main/java/com/google/opensesame/servlets/ProjectServlet.java

```diff
@@ -7,54 +7,117 @@
+    if (parameterValue == null) {
+      return Boolean.valueOf(false);
+    }
+
+    if (parameterValue.equalsIgnoreCase("true") || parameterValue.equalsIgnoreCase("false")) {
+      return Boolean.valueOf(parameterValue);
+    } else {
+      return null;
     }
-    return value;
+  }
+
+  /**
+   * Creates a collection of ProjectData from a collection of ProjectEntities.
+   *
+   * @param projectEntities
+   * @param getFullData
+   * @param getPreviewData
+   * @return Returns the created collection of ProjectData.
+   * @throws IOException Throws if there was an error retrieving the GitHub repository associated
+   *     with a ProjectEntity.
+   */
+  private static Collection<ProjectData> createProjectData(
+      Collection<ProjectEntity> projectEntities, boolean getFullData, boolean getPreviewData)
+      throws IOException {
+    List<ProjectData> projectData = new ArrayList<ProjectData>();
+    for (ProjectEntity projectEntity : projectEntities) {
+      ProjectData newProjectData = new ProjectData(projectEntity);
+      // TODO(Richie): Is there a better way to do this than an if-else block?
+      if (getFullData) {
+        newProjectData.withFullData();
+      } else if (getPreviewData) {
+        newProjectData.withPreviewData();
+      } else {
+        newProjectData.withName();
+      }
+
+      projectData.add(newProjectData);
+    }
+
+    return projectData;
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String projectId = getParameter(request, PROJECT_ID_PARAM, null);
-    if (projectId == null) {
+    Boolean getFullData = booleanFromRequest(request, FULL_DATA_PARAM);
+    if (getFullData == null) {
       ErrorResponse.sendJsonError(
           response,
-          "Must supply project ID.",
+          "Invalid boolean value for the " + FULL_DATA_PARAM + " parameter",
           HttpServletResponse.SC_BAD_REQUEST,
-          "Failed to retrieve project data.");
-      return;
+          "Failed to interpret the request. Please try again later.");
     }
-
-    ProjectEntity projectEntity = ofy().load().type(ProjectEntity.class).id(projectId).now();
-    if (projectEntity == null) {
+    Boolean getPreviewData = booleanFromRequest(request, PREVIEW_DATA_PARAM);
+    if (getPreviewData == null) {
       ErrorResponse.sendJsonError(
           response,
-          "A project with the specified project ID '" + projectId + "' does not exist.",
+          "Invalid boolean value for the " + PREVIEW_DATA_PARAM + " parameter",
           HttpServletResponse.SC_BAD_REQUEST,
-          "Failed to retrieve project data.");
-      return;
+          "Failed to interpret the request. Please try again later.");
+    }
+
+    String[] projectIds = request.getParameterValues(PROJECT_ID_PARAM);
+    Collection<ProjectEntity> projectEntities;
+    if (projectIds != null) {
+      // Get projects by ID if IDs were specified.
```

Comment:
Not sure this comment is necessary. Your variable names and logic here are already telling me what you are doing.

Reply:
Good point, will remove. Thanks!